### PR TITLE
Support ndjson dashboards in beats importer

### LINKED ISSUE #1536: mage importBeats fails to import dashboards in ndjson format
After https://github.com/elastic/beats/pull/27220/, dashboards in Beats modules are stored as Kibana saved objects in ndjson format, current import beats script fails to import these dashboards:
```
2021/08/17 09:18:27 creating from logs source failed: migrating dashboard file failed (path: ../beats/filebeat/module/apache/_meta/kibana/7/dashboard/Filebeat-apache.ndjson): preparing file failed: unmarshalling dashboard file failed: invalid character '{' after top-level value
exit status 1
```

### PR DESCRIPTION
**Pull request by jsoriano**
## What does this PR do?

Add support for ndjson dashboards in beats importer.

Implementation for old format used kibana APIs to convert to saved objects and then it applied some transformations on these objects. Implementation added here applies the transformations directly on the objects in the ndjson files.

## Related issues

- Fixes #1536.

### CODE REVIEW

File: dev/import-beats/kibana.go [hunk 1]
```diff
@@ -216,6 +204,24 @@ func createKibanaContent(kibanaMigrator *kibanaMigrator, modulePath string, modu
 	return kibana, nil
 }
 
+func extractDashboard(kibana *kibanaMigrator, path string, module string, dataStreams []string) (map[string]map[string][]byte, map[string]string, error) {
+	dashboardFile, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "reading dashboard file failed (path: %s)", path)
+	}
+
+	if filepath.Ext(path) == ".ndjson" {
```
Comment (kvch) [inline dev/import-beats/kibana.go:213]: Why is this needed? We do not ship with dashboards json extensions anymore. Have you encountered any? If yes, that is a mistake on my side.

Comment (jsoriano) [reply dev/import-beats/kibana.go:213]: This is only needed to keep the support for the previous format, not needed for current Beats.

We may decide to remove support for previous format, but I don't see any special reason for that, I prefer to keep compatibility as a general rule :)

**Review by kvch — APPROVED** [review]

**Comment by jsoriano** [comment]
Failures were related to https://github.com/elastic/beats/issues/27299, merging.
